CONTRIBUTING.rst

```diff
@@ -67,8 +67,27 @@ Include the following in your patch:
 .. _pre-commit: https://pre-commit.com
 
 
-First time setup
-~~~~~~~~~~~~~~~~
+First time setup using GitHub Codespaces
```

Comment:
I opted to place this in front of the "First time setup in your local environment" section in the hopes that contributors don't go through all the local environment setup steps, and then realize that a codespace may have been a simpler option.

Reply:
Is there a way to move from Codespaces to the contributor's local VS Code? Especially at conference sprints, it would be handy for people to get their own local dev environment as well.

Reply:
The best way for contributors to work in their local dev environments is to follow the local setup steps in your contributing guide.  We currently do not support the use of local resources for Codespaces, which have cloud storage and compute and are designed to be a one-click, isolated, and ephemeral development experience.

Of course, any changes users push from their codespace to their remote fork are available for cloning locally. Contributors may also [export their codespace changes to another branch](https://docs.github.com/en/codespaces/troubleshooting/exporting-changes-to-a-branch), if desired.

Reply:
Isn't there some VS Code feature that allows switching between local and cloud editors? Could have sworn I'd seen something about that.

Reply:
Oh, of course! I forgot to add that in, and apologies if I misunderstood your question.

Yes, you can use Codespaces both in the browser and on desktop. The [GitHub Codespaces](https://marketplace.visualstudio.com/items?itemName=GitHub.codespaces) extension for VS Code works for both. Contributors can change their default editor under their [Codespaces settings](https://docs.github.com/en/codespaces/customizing-your-codespace/setting-your-default-editor-for-github-codespaces).

Would you like me to add that information to the contributing guide?

Reply:
Sure, seems useful to know.